### PR TITLE
ngsolve and friends: update to current versions, fix build

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update everything, fix the build.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.2.1
Xcode 15.1

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by reneeotten** [comment]
as usual, please refrain from making non-functuonal changes (i.e., there is no requirement that PortGroups are sorted alphabetically and this no need to  change things, especially in ports you don't maintain). We'll give @MarcusCalhoun-Lopez some time to look at this.

**Comment by barracuda156** [comment]
@reneeotten I am not gonna insist on these, of course, especially given these are not my ports, but I was asked a several times specifically to make such a sorting by @mascguy – and these should not be controversial. (I do realize they are ”useless“, but so is alignment to 4 spaces or anything at all related to how the text looks.)

Yet we can drop those commits if you guys prefer.

**Comment by mascguy** [comment]
> @reneeotten I am not gonna insist on these, of course, especially given these are not my ports, but I was asked a several times specifically to make such a sorting by @mascguy – and these should not be controversial. (I do realize they are ”useless“, but so is alignment to 4 spaces or anything at all related to how the text looks.)
> 
> Yet we can drop those commits if you guys prefer.

Yes, I do support making such changes. And I'd prefer to keep them, given a choice.

**Comment by reneeotten** [comment]
Alignment with four spaces is what the project has decided on and is stated in the modeline. Nowhere is there a guideline to sort PortGroups alphabetically so that change is more useless and not needed.

**Comment by mascguy** [comment]
> Alignment with four spaces is what the project has decided on and is stated in the modeline. Nowhere is there a guideline to sort PortGroups alphabetically so that change is more useless and not needed.

Actually @ryandesign recommended alphabetizing portgroups some months ago.

So yes, this is an advocated practice, even if isn't formally documented at this point.

**Comment by reneeotten** [comment]
> > Alignment with four spaces is what the project has decided on and is stated in the modeline. Nowhere is there a guideline to sort PortGroups alphabetically so that change is more useless and not needed.
> 
> Actually @ryandesign recommended alphabetizing portgroups some months ago.
> 
> So yes, this is an advocated practice, even if isn't formally documented at this point.

if you're a proponent of this, then please document it in the guide

**Comment by pmetzger** [comment]
Even if this is an important thing to do (doubtful), it should be done in distinct commits.

**Comment by barracuda156** [comment]
@pmetzger Hasn’t it been though?

**Comment by barracuda156** [comment]
@pmetzger This should be ready, I believe. Took quite a bit of time to fix.

File: math/netgen/files/patch-CMakeLists.txt.diff [hunk 1]
```diff
@@ -8,7 +8,7 @@ index 0a05ebb1..677d586b 100644
  
  set(NG_INSTALL_SUFFIX netgen CACHE STRING "Suffix appended to install directories (project name)")
  
--if(APPLE)
+-if(APPLE AND NOT EMSCRIPTEN)
 +if(APPLE AND FALSE)
```
Comment (pmetzger) [inline math/netgen/files/patch-CMakeLists.txt.diff:12]: Isn't X && FALSE always FALSE?

Comment (pmetzger) [reply math/netgen/files/patch-CMakeLists.txt.diff:12]: @barracuda156 ?

Comment (barracuda156) [reply math/netgen/files/patch-CMakeLists.txt.diff:12]: @pmetzger I think I merely adapted the patch we had. Can be changed.

Comment (barracuda156) [reply math/netgen/files/patch-CMakeLists.txt.diff]: @pmetzger Updated to use just `FALSE`.

**Pull request merged**